app/services/manage_action.rb

```diff
@@ -10,7 +11,16 @@ def initialize(params)
   end
 
   def create
-    return previous_action if previous_action.present?
+    unless page.allow_duplicate_actions?
+      return previous_action if previous_action.present?
```

Comment:
I find this structure a little confusing... what if you did
```
if !page.allow_duplicate_actions? && previous_action.present?
  return previous_action
end
```
it's just hard for me to parse the unless and then the inline if